shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1332,7 +1343,30 @@ - (void)handleKeyboardNotification:(NSNotification*)notification {
 
   self.targetViewInsetBottom = calculatedInset;
   NSTimeInterval duration = [info[UIKeyboardAnimationDurationUserInfoKey] doubleValue];
-  [self startKeyBoardAnimation:duration];
+
+  // Flag for simultaneous compounding animation calls.
+  // This captures animation calls made while the keyboard animation is currently animating. If the
+  // new animation is in the same direction as the current animation, this flag lets the current
+  // animation continue with an updated targetViewInsetBottom instead of starting a new keyboard
+  // animation. This allows for smoother keyboard animation interpolation.
+  BOOL keyboardWillShow = beginKeyboardFrame.origin.y > keyboardFrame.origin.y;
+  BOOL keyboardAnimationIsCompounding =
+      self.keyboardAnimationIsShowing == keyboardWillShow && _keyboardAnimationVSyncClient != nil;
+
+  // Mark keyboard as showing or hiding.
+  self.keyboardAnimationIsShowing = keyboardWillShow;
+
+  if (!keyboardAnimationIsCompounding) {
+    [self startKeyBoardAnimation:duration];
+  } else {
+    _keyboardSpringAnimation.reset([[SpringAnimation alloc]
+        initWithStiffness:self.keyboardCASpringAnimation.stiffness
+                  damping:self.keyboardCASpringAnimation.damping
+                     mass:self.keyboardCASpringAnimation.mass
+          initialVelocity:self.keyboardCASpringAnimation.initialVelocity
+                fromValue:self.originalViewInsetBottom
+                  toValue:self.targetViewInsetBottom]);
```

Comment:
Emm why do we need this branch?

Reply:
Need it to update the toValue, so the animation is updated without stopping it. Would probably be cleaner to update the SpringAnimation file to allow for updating the spring without recreating it

Reply:
Got you. You can update the spring animation file to make it can change toValue and fromValue^_^